packages/esm-appointments-app/src/appointments/scheduled/scheduled-appointments.component.tsx

```diff
@@ -16,101 +23,165 @@ interface ScheduledAppointmentsProps {
-          {scheduledAppointmentPanels
-            .filter((panel) => panel.config.showForPastDate)
-            .map((panel, i) => {
-              return <Switch key={i} name={panel.config.title} text={t(panel.config.title)} />;
-            })}
-        </ContentSwitcher>
-      )}
-
-      {dateType === 'futureDate' && (
-        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setCurrentTab(name)}>
-          {scheduledAppointmentPanels
-            .filter((panel) => panel.config.showForFutureDate)
-            .map((panel, i) => {
-              return <Switch key={i} name={panel.config.title} text={t(panel.config.title)} />;
-            })}
-        </ContentSwitcher>
-      )}
+      <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setCurrentTab(name)}>
+        {scheduledAppointmentPanels.filter(shouldShowPanel).map((panel) => {
+          return <Switch key={`panel-${panel.name}`} name={panel.config.title} text={t(panel.config.title)} />;
+        })}
+      </ContentSwitcher>
 
       <ExtensionSlot name={scheduledAppointmentsPanelsSlot}>
-        {(extension) =>
-          extension.config.title === currentTab && (
-            <div className={styles.container}>
-              <Extension
-                state={{
-                  date,
-                  appointmentServiceType,
-                  status: extension.config.status,
-                  title: extension.config.title,
-                }}
-              />
-            </div>
-          )
-        }
+        {(extension) => {
+          return (
+            <ExtensionWrapper
+              extension={extension}
+              currentTab={currentTab}
+              appointmentServiceType={appointmentServiceType}
+              date={date}
+              dateType={dateType}
+              showExtensionTab={showExtension}
+              hideExtensionTab={hideExtension}
+            />
+          );
+        }}
       </ExtensionSlot>
     </>
   );
 };
 
+function useAllowedExtensions() {
+  const [allowedExtensions, setAllowedExtensions] = useState<Readonly<Record<string, boolean>>>({});
+
+  const set = useCallback(
+    (value: boolean, key: string) => {
+      setAllowedExtensions({ ...allowedExtensions, [key]: value });
+    },
+    [allowedExtensions, setAllowedExtensions],
+  );
+
+  return {
+    allowedExtensions,
+    showExtension: set.bind(null, true) as (extension: string) => void,
+    hideExtension: set.bind(null, false) as (extension: string) => void,
+  };
+}
+
+function ExtensionWrapper({
+  extension,
+  currentTab,
+  appointmentServiceType,
+  date,
+  dateType,
+  showExtensionTab,
+  hideExtensionTab,
+}: {
+  extension: ConnectedExtension;
+  currentTab: string;
+  appointmentServiceType: string;
+  date: string;
+  dateType: DateType;
+  showExtensionTab: (extension: string) => void;
+  hideExtensionTab: (extension: string) => void;
+}) {
+  const currentConfig = useRef(null);
```

Comment:
We track the extension config, so that we can display or hide the tab as the configuration gets updated, but we use a Ref so that we can compare the value at last render to the value at the next render and avoid unnecessary renders.